src/rbac-forms/rbacAssignUserTags.jsx

```diff
@@ -0,0 +1,229 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Select from 'react-select';
+import { Spinner, Button, Icon, Row, Col, Grid, ButtonGroup } from 'patternfly-react';
+import { intersectionWith } from '../helpers';
+import { __ } from '../global-functions';
+import { GenericPreviewTable } from '../table';
+
+class RbacAssignCompanyTags extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedCategory: null,
+      categoryEntries: {},
+      selectedEntries: {},
+      isLoaded: false,
+      initialAssignments: {},
+    };
+  }
+
+  componentDidMount() {
+    const { users, categories } = this.props;
+    const commonTags = intersectionWith((a, b) => (a.id === b.id), ...users.map(user => user.tags));
+    const tagsNames = {};
+    const selectedEntries = {};
+    commonTags.forEach((tag) => {
+      tagsNames[tag.name.split('/')[2]] = { value: tag.id };
+    });
+    Object.keys(tagsNames).forEach((categoryName) => {
+      selectedEntries[categories.find(category => category.name === categoryName).value] = { ...tagsNames[categoryName] };
```

Comment:
Can you please merge these two foreaches together? I mean isn't `tag.name.split('/')[2]` same as `categoryName`?